styleguide/STYLEGUIDE.md

```diff
@@ -1,50 +1,1090 @@
+* Indent using 4 spaces rather than tabs Be sure to set this preference in Xcode
 
-![Setting page guide column](xcode-page-guide.png)
+* Method braces and other braces (`if`/`else`/`switch`/`while` etc.) always open on the same line as the statement but close on a new line.
+* Tip: You can re-indent by selecting some code (or **Command-A** to select all) and then **Control-I** (or **Editor ▸ Structure ▸ Re-Indent** in the menu). Some of the Xcode template code will have 4-space tabs hard coded, so this is a good way to fix that.
 
-Also check `Including whitespace-only lines` under `Preferences > Text Editing > Editing > Automatically trim trailing whitespace`.
+**Preferred**:
+```swift
+if user.isHappy {
+  // Do something
+} else {
+  // Do something else
+}
+```
 
-### Identifier names
+**Not Preferred**:
+```swift
+if user.isHappy
+{
+  // Do something
+}
+else {
+  // Do something else
+}
+```
 
-Identifier names (e.g. variables, enum values, etc) must have a minimum length of 1 characters and a maximum length of 60 characters.
+* There should be one blank line between methods and up to one blank line between type declarations to aid in visual clarity and organization. Whitespace within methods should separate functionality, but having too many sections in a method often means you should refactor into several methods.
 
-### Type names
+* There should be no blank lines after an opening brace or before a closing brace.
 
-Type names (e.g. classes, structs, etc) must have a minimum length of 3 characters and a maximum length of 100 characters.
+* Closing parentheses should not appear on a line by themselves.
 
-### Multiline function parameters
+**Preferred**:
+```swift
+let user = try await getUser(
+  for: userID,
+  on: connection)
+```
 
-If you separate a function definition or call on to separate lines (because of the line length rule, perhaps) then _each_ parameter should be on a separate line.  Xcode should automatically align them for you.
+**Not Preferred**:
+```swift
+let user = try await getUser(
+  for: userID,
+  on: connection
+)
+```
 
-This won't be enforced by Swift Lint.
+* Colons always have no space on the left and one space on the right. Exceptions are the ternary operator `? :`, empty dictionary `[:]` and `#selector` syntax `addTarget(_:action:)`.
 
-### Logging
+**Preferred**:
+```swift
+class TestDatabase: Database {
+  var data: [String: CGFloat] = ["A": 1.2, "B": 3.2]
+}
+```
 
-We use [unified logging system](https://developer.apple.com/documentation/os/logging) to capture log messages. To send a message to the logging system, call the `os_log` function. Please use debug log level to protect users' privacy.
+**Not Preferred**:
+```swift
+class TestDatabase : Database {
+  var data :[String:CGFloat] = ["A" : 1.2, "B":3.2]
+}
+```
+
+* Long lines should be wrapped at around 70 characters. A hard limit is intentionally not specified.
```

Comment:
70 seems very narrow, especially if most of us use big screens

Reply:
@jaceklyp what's your suggestion to replace 70? I don't have strong opinions here.